stools/clair/__init__.py

```diff
@@ -95,58 +145,78 @@ def help(retval=0):
         version()
         sys.exit(0)
 
+    if args.report is None and args.report_format is not None:
+        print("\nWarning: --report-format has no effect when --report is not passed\n")
+
     # Generate Clair controller
     clair = Clair(args.clair_host, args.clair_port)
     if not clair.ping():
         sys.exit(1)
 
     print(clair)
-    
+
     # Local Server
-    webroot = '/var/www/images'
+    webroot = "/var/www/images"
 
     # Start the server and serve static files from root
-
     if args.server is True:
-        print('\n1. Starting server...')
-        server = 'http://%s:%s/' %(args.host, args.port)
+        print("\n1. Starting server...")
+        server = "http://%s:%s/" % (args.host, args.port)
         process = Process(target=start, args=(args.port, args.host, webroot))
-        # start(port=args.port, host=args.host, static_folder=webroot)
         process.daemon = True
         process.start()
-        time.sleep(1)
+        sleep(1)
 
-    # Check health of server
-    print('\n1. Checking server...')
+    # Check health of clair server
+    print("\n1. Checking server...")
     response = requests.get(server)
     if response.status_code != 200:
-        print('Server not found running at %s' %server)
+        print("Server not found running at %s" % server)
         sys.exit(1)
 
-    print('2. Processing images!')
+    print("2. Processing images!")
     for image in args.images:
 
         # 1. decompress to sandbox --> tar.gz
         targz = export_to_targz(image)
-        print("...exported %s to %s" %(image, targz))
+        print("...exported %s to %s" % (image, targz))
 
         # 2. Move to webroot
-        targz_web = os.path.join(webroot, os.path.basename(targz))
-        os.rename(targz, targz_web)
-        targz_url = "%simages/%s" %(server, os.path.basename(targz))
+        targz_web = join(webroot, basename(targz))
+        rename(targz, targz_web)
+        targz_url = "%simages/%s" % (server, basename(targz))
 
         # 3. Scan with Clair, use image name
-        print("...serving %s to Clair" %(targz_url))
-        clair.scan(targz_web, os.path.basename(image))
+        print("...serving %s to Clair" % targz_url)
+        clair.scan(targz_web, basename(image))
 
         # 4. Generate report
-        print('3. Generating report!')
-        report = clair.report(os.path.basename(image))
-        clair.print(report)
+        print("3. Generating report!")
+        report = clair.report(basename(image))
+        if args.report is not None:
+            fpath = join(
+                args.report, splitext(basename(image))[0] + ".%s" % args.report_format
```

Comment:
Same here, and refactor so report is always json.

Reply:
Will do. Just thought it might be useful :+1:

Reply:
Reverted :+1: